core/drivers/stm32_rtc.c

```diff
@@ -1063,6 +1080,17 @@ static TEE_Result stm32_rtc_probe(const void *fdt, int node,
 		if (res)
 			return res;
 
+		if (!rtc_dev.itr_chip)
+			return TEE_ERROR_GENERIC;
+
+		res = interrupt_create_handler(rtc_dev.itr_chip,
+					       rtc_dev.itr_num,
+					       stm32_rtc_it_handler,
+					       &rtc_dev, 0,
```

Comment:
This changes what was done in previous patch, there is a way to make it better, and in the same patch IMO. But it still should depend on secure configuration of features